source/reference/combining.rst

```diff
@@ -251,6 +251,9 @@ subschemas.  The following schema is equivalent to the one above:
       ]
     }
 
+.. warning::
+   Careful consideration should be taken when using multiple `oneOf` entries with nested data structures as the number of validation checks performed will increase exponentially resulting in poor performance.
+
```

Comment:
As I mentioned in https://github.com/json-schema-org/understanding-json-schema/issues/130#issuecomment-1791469674, the issue isn't with _multiple_ `oneOf`s.  The performance hit is caused by the nature of `oneOf` requiring that all subschemas be evaluated.

> the number of validation checks performed will increase exponentially

This isn't strictly true (mathematically).  It's a linear increase.

```yaml
oneOf:
  - oneOf:
    - oneOf:
      - a
      - b
      - c
    - d
  - e
```

This still only needs to check 7 subschemas (`a`-`e` + the two inner `oneOf`s).  However, a problem does crop up when one of those subschemas references back to the root.  It's this recursive referencing that can result in a more-than-linear increase.